cmd/node/main.go

```diff
@@ -542,6 +552,70 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		log.Debug("no epoch db found in storage", "error", errNotCritical.Error())
 	}
 
+	epochFoundInStorage := errNotCritical == nil
+
+	shouldCallEpochStartDataProvider := bootstrap.ShouldSyncWithTheNetwork(
+		startTime,
+		epochFoundInStorage,
+		nodesConfig,
+		generalConfig,
+	)
+	shouldCallEpochStartDataProvider = true
+	if shouldCallEpochStartDataProvider {
+		networkComponents, err = factory.NetworkComponentsFactory(p2pConfig, log, &blake2b.Blake2b{})
+		if err != nil {
+			return err
+		}
+
+		err = networkComponents.NetMessenger.Bootstrap()
+		if err != nil {
+			return err
+		}
+		time.Sleep(2 * time.Second)
+
+		marshalizer := &marshal.JsonMarshalizer{}
+		hasher := &blake2b.Blake2b{}
+		simpleNodesConfigProvider := nodesconfigprovider.NewSimpleNodesConfigProvider(nodesConfig)
+		metaBlockInterceptor, err := bootstrap.NewSimpleMetaBlockInterceptor(marshalizer, hasher)
+		if err != nil {
+			return err
+		}
+		shardHdrInterceptor := bootstrap.NewSimpleShardHeaderInterceptor(marshalizer)
+		metaBlockResolver, err := bootstrap.NewSimpleMetaBlocksResolver(networkComponents.NetMessenger, marshalizer)
+		epochStartDataArgs := bootstrap.ArgsEpochStartDataProvider{
+			Messenger:              networkComponents.NetMessenger,
+			Marshalizer:            marshalizer,
+			Hasher:                 hasher,
+			NodesConfigProvider:    simpleNodesConfigProvider,
+			MetaBlockInterceptor:   metaBlockInterceptor,
+			ShardHeaderInterceptor: shardHdrInterceptor,
+			MetaBlockResolver:      metaBlockResolver,
+		}
+		epochRes, err := bootstrap.NewEpochStartDataProvider(epochStartDataArgs)
```

Comment:
can you move everything inside epochStart / bootstrap ?

Reply:
done. Now main only has the factory init, the Create() method call and the Bootstrap() method call